adr/0004-supported-python-versions.md

```diff
@@ -0,0 +1,30 @@
+# Supported Python Versions
+
+Date: 2017-11-20
+
+## Status
+
+Proposed
+
+## Context 
+
+eLife has numerous projects written completely and partly with the [Python programming language](https://www.python.org/).
+
+In order to provide language version consistency across projects we need to get a consensus on which versions we are going to support. 
+
+We have only gone up to Python 3.5 due to the default Python versions pre installed on the Ubuntu distributions we use.
+
+## Decision
+
+We will use Python >=2.7.14 as our default version for any project that solely uses or supports Python 2.
+
+We will use Python 3.5 as our default supported version for any project that solely uses or supports Python 3.
+
+## Consequences
+
+Any libraries that are currently only supporting Python 2 are to be upgraded to support both Python 2 & 3.
+
+Any projects that are currently using Python 2 are to be upgraded to Python 3.
```

Comment:
Contradicts the decision above. I guess some projects will remain on Python 2 where they use external libraries that don't support Python 3?

Reply:
We have to maintain support for Python 2 for existing libraries, not for us long term, as assumably in the near future all projects will be upgraded from 2 to 3, but for the transitional period. Any new projects or libraries can be Python 3 only. 

If a library or a project that we maintain is to remain active then it has to adhere to this, otherwise, it has a 2020 expiry date. 

If one of our libraries or projects currently depend on something that only supports Python 2 then that can be addressed by either finding alternatives, writing our own implementation (if it makes sense) or possibly forking if need be but it should not be the reason we cannot give a library or project Python 3 support.